frontend/src/pages/UnitDetails.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useState } from "react";
+import { Helmet } from "react-helmet-async";
+import { useParams } from "react-router";
+import { Link } from "react-router-dom";
+import styled from "styled-components";
+
+import { Unit, getUnit } from "@/api/units";
+import { Page } from "@/components";
+import { Button } from "@/components/Button";
+
+export function UnitDetails() {
+  const [unit, setUnit] = useState<Unit>();
+  const { id } = useParams();
+
+  React.useEffect(() => {
+    if (id !== undefined) {
+      void getUnit(id).then((result) => {
+        if (result.success) {
+          setUnit(result.data);
+        }
+      });
+    }
+  }, []);
+
+  const Row = styled.div`
+    display: flex;
+    flex-direction: row;
+    background-color: #fbf7f3;
+  `;
+
+  const Column = styled.div`
+    display: flex;
+    flex-direction: column;
+    background-color: #fbf7f3;
+    jusify-content: evenly-spaced;
+    gap: 16px;
+  `;
+
+  const DetailsRow = styled(Row)`
+    gap: 20px;
+  `;
+
+  const SectionColumn = styled(Column)`
+    width: 50%;
+  `;
+
+  const MainColumn = styled(Column)`
+    padding: 10%;
+  `;
+
+  const GapColumn = styled(Column)`
+    padding: 10%;
+  `;
+
+  const RentPerMonth = styled.h1`
+    font-size: 48px;
+    font-family: "Neutra Text";
+    font-weight: 700;
+    line-height: 150%;
+    line-spacing: 0.96px;
+
+    margin: 0;
+  `;
+
+  const Header = styled.div`
+    font-size: 32px;
+    margin: 0;
+    font-weight: 700;
+    font-family: "Neutra Text";
+    line-height: 150%;
+    line-spacing: 0.64px;
+    margin-top: 32px;
+  `;
+
+  const Text = styled.div`
+    font-family: "Montserrat"
+    font-weight: 400;
+    font-size: 20px;
+    line-height: 150%
+    letter-spacing: 0.4px;
+
+  `;
+
+  const List = styled.ul`
+    margin-top: 0;
+    margin-bottom: 0;
+  `;
+
+  const StrongText = styled(Text)`
+    font-weight: 600;
+  `;
+
+  const ListText = styled.li`
+    font-family: "Montserrat"
+    font-weight: 400;
+    font-size: 20px;
+    line-height: 150%
+    letter-spacing: 0.4px;
+  `;
+
+  const Address = styled(Header)``;
+
+  const ButtonPadding = styled.div`
+    display: flex;
+    flex-direction: row;
+    padding: 20px 10% 0px 10%;
+  `;
+
+  const DoesNotExist = styled.h1`
+    font-size: 48px;
+    font-family: "Neutraface Text", sans-serif;
+    font-weight: 550;
+    line-height: 150%;
+    line-spacing: 0.96px;
+    padding: 20px 10% 20px 10%;
+  `;
```

Comment:
We should move these styled components outside of the component function. Having styled components within the render method actually hurts performance ([explanation from the docs](https://styled-components.com/docs/faqs#why-should-i-avoid-declaring-styled-components-in-the-render-method)).